src/extensions/default/InlineColorEditor/main.js

```diff
@@ -39,41 +41,29 @@ define(function (require, exports, module) {
      * @return {?{color:String, marker:TextMarker}}
      */
     function prepareEditorForProvider(hostEditor, pos) {
-        var colorRegEx, cursorLine, match, sel, start, end, endPos, marker;
-
-        sel = hostEditor.getSelection();
-        if (sel.start.line !== sel.end.line) {
-            return null;
-        }
-
-        colorRegEx = new RegExp(ColorUtils.COLOR_REGEX);
+        var cursorLine, cursorLine2, marker, endPos, end;
         cursorLine = hostEditor.document.getLine(pos.line);
 
-        // Loop through each match of colorRegEx and stop when the one that contains pos is found.
-        do {
-            match = colorRegEx.exec(cursorLine);
-            if (match) {
-                start = match.index;
-                end = start + match[0].length;
+        // Make a copy of cursorLine after removing spaces and ":" so that we can check for it in properties
+        cursorLine2 = cursorLine.replace(/\s/g,'');
```

Comment:
does `cursorLine` get reconsulted later? If not, there is no need to define a `cursorLine2` here, just recycle `cursorLine`. Alternatively, it might make more sense to name this variable after what you're actually grabbing:
```
cssPropertyName = cursorLine.split(':')[0].trim();
```
and then use that in the `properties[cssPropertyName]` check.